docs/usage/billing.md

```diff
@@ -1,74 +1,164 @@
+| --------------------- | ---------------------------- | :-------: | :-----------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `amount`              | `number`                     |    Yes    |       -       | The amount to charge                                                                                                                                                                           |
+| `currencyCode`        | `string`                     |    Yes    |       -       | The currency to charge, currently only `"USD"` is accepted                                                                                                                                     |
+| `interval`            | `BillingInterval`            |    Yes    |       -       | `BillingInterval` value                                                                                                                                                                        |
+| `trialDays`           | `number`                     |    No     |       -       | Give merchants this many days before charging. Not available for `OneTime` plans                                                                                                               |
+| `replacementBehavior` | `BillingReplacementBehavior` |    No     |       -       | `BillingReplacementBehavior` value, see [the reference](https://shopify.dev/api/admin-graphql/2022-07/mutations/appSubscriptionCreate) for more information. Not available for `OneTime` plans |
 
 ## Checking for payment
 
-The main method for billing is `Shopify.Billing.check`.
-This method will take in the following parameters:
+Once the app has an access token, you can call `shopify.billing.check` to look for an existing payment.
+
+Here's a typical example of how apps might do that:
+
+```ts
+// This can happen at any point after the merchant goes through the OAuth process
+function billingMiddleware(req, res, next) {
+  const session = shopify.session.getCurrent({
+    isOnline: true,
+    rawRequest: req,
+    rawResponse: res,
+  });
+
+  const hasPayment = await shopify.billing.check({
+    session,
+    plans: ['My billing plan'],
+    isTest: true,
+  });
+
+  if (hasPayment) {
+    next();
+  } else {
+    // Either request payment now or redirect to plan selection page
+  }
+}
+
+app.use('/requires-payment/*', billingMiddleware);
+```
+
+**Note**: this method will always query Shopify's API because merchants can cancel subscriptions at any point. Depending on the number of requests your app handles, you might want to cache a merchant's payment status, but you should make sure to periodically call this method to ensure you're blocking unpaid access.
```

Comment:
```suggestion
**Note**: the `check` method will always query Shopify's API because merchants can cancel subscriptions at any point. Depending on the number of requests your app handles, you might want to cache a merchant's payment status, but you should periodically call this method to ensure you're blocking unpaid access.
```